storage/src/error.rs

```diff
@@ -0,0 +1,121 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use std::{fmt, io};
+use thiserror::Error;
+
+/// Storage error kind.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum StorageErrorKind {
+    /// The target index does not exists
+    DoesNotExist,
+    /// The request credentials do not allow for this operation.
+    Unauthorized,
+    /// A third-party service forbids this operation.
+    Service,
+    ///
```

Comment:
being nitty about this empty comment. `InternalError` is self describing thought